src/App.tsx

```diff
@@ -1,14 +1,48 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as todoActions } from './features/todos';
+import { filteredTodos } from './utils/filteredTodos';
+import { warningTimer } from './utils/warningTimer';
 
 export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState('');
```

Comment:
> the name of state speaks that it should has a boolean value, but you keep an error message there actually, so plz reduce it to one point

As usual, prefix `has` also uses for boolean type)
Usually, `error` has name `error` =)
```suggestion
  const [error, setError] = useState('');
```